lumicks/pylake/fdcurve.py

```diff
@@ -79,6 +80,31 @@ def __getitem__(self, item):
         new_curve.stop = new_curve.f._src.stop
         return new_curve
 
+    def _slice_by_distance(self, min_dist, max_dist, max_gap=3):
+        """Return a subset of this FD curve within a selected distance range"""
+        # logical mask of data lying within selected distance range
+        mask_in_range = np.logical_and(min_dist <= self.d.data, self.d.data <= max_dist).astype(np.bool)
+
+        # find runs of data that are not in selected distance range
+        # runs shorter than the allowed gap should be allowed
+        ranges, lengths = find_contiguous(~mask_in_range)
+        for rng, ln in zip(ranges, lengths):
+            if ln < max_gap:
+                # ignore first and last runs of data, as total length is not defined
+                if rng[0] == 0 or rng[1] == self.d.data.size:
+                    continue
+                mask_in_range[slice(*rng)] = True
+
+        # find longest contiguous run of data and return corresponding FD curve
+        ranges, lengths = find_contiguous(mask_in_range)
+        longest_range = ranges[np.argmax(lengths)]
+        start = self.d.timestamps[longest_range[0]]
+        try:
+            stop = self.d.timestamps[longest_range[-1]]
+            return self[start:stop]
+        except IndexError: # slicing to the end of data
+            return self[start:]
```

Comment:
Can we make this a bit more specific? Why not just use an if-statement that checks whether the end of the range is beyond the end of the timestamps? The `try` potentially catches more `IndexErrors` that may go silently unnoticed now.